app/src/composables/use-mutex.ts

```diff
@@ -0,0 +1,54 @@
+import { sleep } from '@directus/sdk';
+
+const MutexKey = ['auth_refresh'] as const;
+
+type MutexKey = (typeof MutexKey)[number];
+
+export function useMutex(key: MutexKey, expiresMs: number) {
+	const internalKey = `directus-mutex-${key}`;
+	const useWebLock = window.isSecureContext;
```

Comment:
Should be in order to actually do feature detection if the browser supports locks.
`navigator.locks` is `undefined` in case of a unsecure context.

```suggestion
	const useWebLock = !!navigator.locks;
```